### ISSUE TITLE
Acquire WIFI lock too

### ISSUE THREAD

**Issue by bkhall** [brianwernick/ExoMedia#76]
When setting wake mode, consider also getting a wifi lock if the datasource is a remote URI.

**Comment by brianwernick** [comment]
What area of the library are you talking about?  EMVideoView, EMAudioPlayer, EMPlaylistService?

**Comment by bkhall** [comment]
Well, in the particular app I'm writing, we are using your EMAudioPlayer to stream an mp3 icecast feed.

No to worry though, we grab our own wifi and wake locks. I was just noticing that your library has the wake lock capability too, but could possibly also use a wifi lock for remote streaming, in whatever widget it applies to.

**Comment by brianwernick** [comment]
Hmm, OK.  I'll think about it some more before I give a definitive answer either way...

**Comment by bkhall** [comment]
Yep, that's all my comment was meant for - something to think about.

**Comment by brianwernick** [comment]
OK.  I've decided not to handle the WiFi lock in the EMAudioPlayer or EMVideoView because I don't want to worry about handling edge cases between local/web playback and keeping/releasing the locks.

However, I think that the WiFi lock should be updated in the EMPlaylistService to support locking when videos are playing as well.  That said, the change probably won't make it in to ExoMedia (instead it will be in [PlaylistCore][PlaylistCore] once the separation is complete for the 3.0 ExoMedia release). If you are interested in an explanation for the separation see https://github.com/brianwernick/ExoMedia/issues/48#issuecomment-173789504

[PlaylistCore]:https://github.com/brianwernick/PlaylistCore

**Comment by brianwernick** [comment]
I'm going to close this as it is now out of scope on ExoMedia.  I've added this to my list of things to do for [PlaylistCore](https://github.com/brianwernick/PlaylistCore)
